dist/derivable.js.flow

```diff
@@ -0,0 +1,138 @@
+/**
+ * @flow
+ */
+
+export interface Derivable<T> {
+
+  derive<E>(f: (value: T) => E): Derivable<E>;
+  derive(prop: (string | Derivable<string>)): Derivable<mixed>;
+  derive(index: (number | Derivable<number>)): Derivable<mixed>;
+  derive(re: (RegExp | Derivable<RegExp>)): Derivable<Array<string>>;
+  derive<E>(f: Derivable<(value: T) => E>): Derivable<E>;
+  derive(args: Array<mixed>): Array<Derivable<mixed>>;
+  derive<A, E>(f: (value: T, a: A) => E, a: (A | Derivable<A>)): Derivable<E>;
+  derive<A, B, E>(f: (value: T, a: A, b: B) => E, a: (A | Derivable<A>), b: (B | Derivable<B>)): Derivable<E>;
+  derive<E>(f: (value: T, ...args: Array<mixed>) => E, ...args: Array<mixed>): Derivable<E>;
+
+  mDerive<E>(f: (value: T) => E): Derivable<E>;
+  mDerive<A, E>(f: (value: T, a: A) => E, a: (A | Derivable<A>)): Derivable<E>;
+  mDerive<A, B, E>(f: (value: T, a: A, b: B) => E, a: (A | Derivable<A>), b: (B | Derivable<B>)): Derivable<E>;
+  mDerive<E>(f: (value: T, ...args: Array<mixed>) => E, ...args: Array<mixed>): Derivable<E>;
+
+  react(f: (value: T) => void, options?: Lifecycle): void;
+
+  mReact(f: (value: T) => void, options?: Lifecycle): void;
+
+  get(): T;
+
+  is(other: mixed): Derivable<boolean>;
+
+  and(other: mixed): Derivable<mixed>;
+
+  mAnd(other: mixed): Derivable<mixed>;
+
+  or(other: mixed): Derivable<mixed>;
+
+  mOr(other: mixed): Derivable<mixed>;
+
+  then(thenD: mixed, elseD: mixed): Derivable<mixed>;
+
+  mThen(thenD: mixed, elseD: mixed): Derivable<mixed>;
+
+  not(): Derivable<boolean>;
+
+  switch(...args: Array<mixed>): Derivable<mixed>;
+
+  withEquality(equals: (a: mixed, b: mixed) => boolean): Derivable<T>;
+};
+
+export interface Atom<T> extends Derivable<T> {
+
+  set(value: T): void;
```

Comment:
I made `set()` not to change type parameter of the atom. I think this more aligns with how it's going to be used. Also Flow could infer type of:
```
let a = atom(42);
a.set('ok')
```
as `Atom<string | number>` automatically (yay control flow analysis!) automatically!